src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -442,9 +442,22 @@ func renderResultsInstantaneousJSON(
 		jw.EndObject()
 
 		jw.BeginObjectField("value")
+
+		onlynans := true
+		for a := 0; a < length; a++ {
+			if !math.IsNaN(vals.ValueAt(a)) {
+				onlynans = false
+				break
+			}
+		}
```

Comment:
Should this return the last non nan value, or explicitly 0 if there are trailing nans?